collection/lscq/types.go

```diff
@@ -131,6 +131,8 @@ type uint64SCQ struct {
 type scqNodeUint64 struct {
 	flags uint64 // isSafe 1-bit + isEmpty 1-bit + cycle 62-bit
 	data  uint64
+	_     sync.Mutex                              // Only for architectures that do not support 128bit atomic operations
```

Comment:
I think we don't need to support architectures that don't support 128-bit CAS. Adding a `sync.Mutex` usually comes with performance losses. It would be nice if you could run the benchmark to see the diff :)